src/state/pools/index.ts

```diff
@@ -42,16 +42,20 @@ export const PoolsSlice = createSlice({
 export const { setPoolsPublicData, setPoolsUserData, updatePoolsUserData } = PoolsSlice.actions
 
 // Thunks
-export const fetchPoolsPublicDataAsync = () => async (dispatch) => {
+export const fetchPoolsPublicDataAsync = (currentBlock) => async (dispatch) => {
   const blockLimits = await fetchPoolsBlockLimits()
   const totalStakings = await fetchPoolsTotalStaking()
 
   const liveData = poolsConfig.map((pool) => {
     const blockLimit = blockLimits.find((entry) => entry.sousId === pool.sousId)
     const totalStaking = totalStakings.find((entry) => entry.sousId === pool.sousId)
+    const isPoolFinished =
+      pool.isFinished || (currentBlock > 0 && blockLimit ? currentBlock > Number(blockLimit.endBlock) : false)
```

Comment:
This condition is a bit confusing, can we try and improve readability? One step would be to break out the second part into one or more variables to help describe what each does?

```
currentBlock > 0 && blockLimit ? currentBlock > Number(blockLimit.endBlock) : false
```

Reply:
done